test/extensions/secondorder/hbp/test_kfac.py

```diff
@@ -23,3 +31,48 @@ def test_kfac_not_supported(problem):
         BackpackExtensions(problem).kfac()
 
     problem.tear_down()
+
+
+@pytest.mark.montecarlo
+@pytest.mark.parametrize("problem", BATCH_SIZE_1_PROBLEMS, ids=BATCH_SIZE_1_IDS)
+def test_kfac_should_approx_ggn_montecarlo(problem):
+    """Check that for batch_size = 1, the K-FAC is the same as the GGN.
+
+    Args:
+        problem (ExtensionsTestProblem): Test case.
+    """
+    problem.set_up()
+    # calculate GGN
+    autograd_res = AutogradExtensions(problem).ggn_blocks()
+
+    # calculate backpack average
+    mc_samples = 300000
+    backpack_kfac = BackpackExtensions(problem).kfac(mc_samples)
+    backpack_res = [kfacs_to_mat(kfac) for kfac in backpack_kfac]
+
+    # check the values
+    check_sizes_and_values(autograd_res, backpack_res, atol=1e-1, rtol=1e-1)
```

Comment:
Maybe reduce both to `1e-3`

Reply:
TODO: Verify it works on GPU

Reply:
Reduced tolerances in commit
"Fixed shape of ggn blocks, type annotation, tol sizes, and commenting"

Reply:
Regarding running on the GPU: as written currently, it did in fact not pass on the GPU. 

I looked back at the tests I ran to determine when it passed, and found an issue with a simplifying assumption I made.
Original procedure: for each testing threshold (exp: 1e-1, 1e-2, 1e-3, etc), gradually increase the number of mc_samples until the test passes, then record this as the minimum number of mc_samples required.

Incorrect assumption: if a test has passed, adding more samples will also result in the test passing.

After looking at all passed mc_samples for each tol threshold, it seems that (probably intuitively) this is not the case, and it actually enters a period of pass/fail oscillation before consistently passing as the number of samples increases.

**That being said, GPU vs CPU performance differs slightly in the exact conversion, but seems to be comparable for this problem.**

**As far as tolerance, for 3e5 samples, 1e-2 may be safer. If 1e-3 tolerance is preferred, 3e6 samples would be safer**

And I would just like to be explicit that I only tried this on the problem outlined in kfac_settings.py (seeded), so a more accurate analysis of the relationship between mc_samples and tolerance would require another round of testing (which of course I can do, if you feel it would be valuable).

Reply:
Thanks for the detailed analysis. Indeed, increasing the samples will only improve the accuracy on average, but their can surely be individual outliers.

How about using `5e-3`, which lies between `1e-2` and `1e-3`?

Reply:
Changed to 5e-3 in commit
"Some docstring and code cleanup"